daemon/daemon.go

```diff
@@ -807,6 +807,7 @@ func NewDaemon(config *config.Config, registryService registry.Service, containe
 			imageStore: is,
 			layerStore: ls,
 		}
+		d.stores[runtime.GOOS] = *d.deltaStore
```

Comment:
@robertgzr, as per Flowdock discussions and given that issue #98 includes steps to reproduce the problem, it would be nice for the PR to include an integration test that more or less reproduces those steps and fails before your changes, and succeeds after your changes. :)
To run the existing tests, I was using Docker on my Mac and the Makefile at the root of the repo:
```
make test-unit
make test-integration
```
Also for iterative development and to run individual tests, there are some helpful comments at the top of the Dockerfile (at the root of the repo):
```
make balena
docker run -v `pwd`:/go/src/github.com/docker/docker --privileged -it <image_name> /bin/bash
docker run -e DOCKER_GITCOMMIT=foo --privileged <image_name> hack/make.sh test-integration test-docker-py
```

Reply:
@pdcastro looking into how integration tests work here I am bit lost...

Going by https://github.com/resin-os/balena-engine/blob/17.12-resin/integration/doc.go#L2 the tests assume a running daemon. To reproduce the issue I think we woud need to be able to restart the daemon with new configs, not to mention we would need a registry somewhere since we also need to pull the delta in from somewhere